beebop/assignClusters.py

```diff
@@ -27,93 +33,421 @@ def hex_to_decimal(sketches_dict) -> None:
+    :return tuple[list, list]: [list initial not found sample hashes,
+        list of clusters assigned to initial not found samples]
+    """
+    not_found_sketches_dict = {
+        key: value
+        for key, value in sketches_dict.items()
+        if key in not_found_query_names
+    }
+    sketch_to_hdf5(not_found_sketches_dict, output_full_tmp)
+
+    assign_query_clusters(
+        config, config.full_db_fs, not_found_query_names, output_full_tmp
+    )
+    not_found_query_names_new, not_found_query_clusters, _, _, _, _, _ = (
+        summarise_clusters(
+            output_full_tmp,
+            config.species,
+            config.full_db_fs.db,
+            not_found_query_names,
+        )
+    )
+
+    copy_include_files(output_full_tmp, config.out_dir)
+    merge_partial_query_graphs(config.p_hash, config.fs)
+
+    return not_found_query_names_new, not_found_query_clusters
+
+
+def update_external_clusters(
+    config: ClusteringConfig,
+    not_found_query_names: list,
+    external_clusters: dict,
+    previous_query_clustering: str,
+) -> None:
+    """
+    [Updates the external clusters with the external clusters found
+    in the new previous query clustering from assigning
+    using the full database.
+    This function reads the external clusters from the
+    new previous query clustering file
+    and updates the initial external clusters file with the clusters
+    that were not found in the current query with the full database.]
+
+    :param config: [ClusteringConfig
+        with all necessary information]
+    :param not_found_query_names: [list of sample hashes
+        that were not found]
+    :param external_clusters: [dict of sample hashes
+        to external cluster labels]
+    :param previous_query_clustering: [path to previous
+        query clustering file]
+    """
+    not_found_prev_querying = config.fs.external_previous_query_clustering_tmp(
+        config.p_hash
+    )
+    external_clusters_not_found, _ = get_external_clusters_from_file(
+        not_found_prev_querying,
+        not_found_query_names,
+        config.external_clusters_prefix,
+    )
+    update_external_clusters_csv(
+        previous_query_clustering,
+        not_found_query_names,
+        external_clusters_not_found,
+    )
+    external_clusters.update(external_clusters_not_found)
+
+
+def merge_partial_query_graphs(p_hash: str, fs: PoppunkFileStore) -> None:
+    """
+    [Merge the partial query graphs from the full assign job with the
+    partial query graphs from the initial assign job.]
+
+    :param p_hash: [project hash]
+    :param fs: [PoppunkFileStore with paths to input files
+    """
+    tmp_assign_subset_file = fs.partial_query_graph_tmp(p_hash)
+    main_subset_file = fs.partial_query_graph(p_hash)
+    with open(tmp_assign_subset_file, "r") as f:
+        failed_lines = set(f.read().splitlines())
+    with open(main_subset_file, "r") as f:
+        main_lines = set(f.read().splitlines())
```

Comment:
As discussed, maybe do some renaming here, as "failed_lines" shouldn't be failed anymore. 
Maybe "from_full_db" and "from_refs_db" or something..